packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -87,35 +95,46 @@ type RaceStatusResult = {
 
 type GetPurchaseConfigArgs = {
   contentId: ID
-  contentType: ContentType
+  contentType: PurchaseableContentType
 }
 
 function* getContentInfo({ contentId, contentType }: GetPurchaseConfigArgs) {
-  if (contentType !== ContentType.TRACK) {
-    throw new Error('Only tracks are supported')
-  }
-
-  const trackInfo = yield* select(getTrack, { id: contentId })
+  const metadata =
+    contentType === PurchaseableContentType.ALBUM
+      ? yield* select(getCollection, { id: contentId })
+      : yield* select(getTrack, { id: contentId })
   const purchaseConditions =
-    trackInfo?.stream_conditions ?? trackInfo?.download_conditions
+    metadata?.stream_conditions ??
+    (metadata && 'download_conditions' in metadata
+      ? metadata?.download_conditions
+      : null)
```

Comment:
nit: does `metadata?.stream_conditions ?? metadata?.download_conditions` not work? does it need to be `null`?

Reply:
functionally fine, but typescript doesn't know that `download_conditions` is ok to access on the object without the `in` check first to restrict the union type